slides/k8s/operators.md

```diff
@@ -0,0 +1,366 @@
+# Operators
+
+- Operators are one of the many ways to extend Kubernetes
+
+- We will define operators
+
+- We will see how they work
+
+- We will install a specific operator (for ElasticSearch)
+
+- We will use it to provision an ElasticSearch cluster
+
+---
+
+## What are operators?
+
+- Operators combine two things:
+
+  - Custom Resource Definitions
+
+  - controller code watching the corresponding resources and acting upon them
+
+- A given operator can define one or multiple CRDs
+
+- The controller code (control loop) can run within the cluster or elsewhere
```

Comment:
This might need a little more clarification - it's quite open-ended!

Reply:
Indeed. New versions reads:

- The controller code (control loop) typically runs within the cluster
  (running as a Deployment with 1 replica is a common scenario)
- But it could also run elsewhere
  (nothing mandates that the code run on the cluster, as long as it has API access)

Reply:
Much better! Thanks.

Reply:
<3